builder/context.go

```diff
@@ -28,40 +29,53 @@ var (
 func (b *Builder) getDockerRunArgs(
 	volName string,
 	stepWorkDir string,
-	step *graph.Step) []string {
-	args := []string{"docker", "run"}
+	step *graph.Step,
+	envs []string,
+	entrypoint string,
+	cmd string) []string {
+
+	var args []string
+	// Run user commands from a shell instance in order to mirror the shell's field splitting algorithms,
+	// so we don't have to write our own argv parser for exec.Command.
+	if runtime.GOOS == "windows" {
+		// TODO: finalize on how to handle the Windows shell.
```

Comment:
likely need to enclose the command in {}

Reply:
will make a separate issue for windows support